src/bundle/Resources/public/js/scripts/fieldType/base/base-rich-text.js

```diff
@@ -398,17 +398,31 @@
             const links = container.querySelectorAll('a');
             const anchorPrefix = '#';
             const protocolPrefix = 'http://';
+            const restrictedKeywords = ['ezcontent', 'ezlocation'];
 
             links.forEach((link) => {
                 const href = link.getAttribute('href');
                 const protocolPattern = /^https?:\/\//i;
+                const protocolHref = protocolPrefix.concat(href);
 
-                if (href && href.indexOf(anchorPrefix) !== 0 && !protocolPattern.test(href)) {
-                    const protocolHref = protocolPrefix.concat(href);
+                if (!href) {
```

Comment:
These four `if`s could be merged into one,
```
if (!href || href.indexOf(anchorPrefix) === 0 || protocolPattern.test(href) || restrictedKeywords.some((keyword) => href.includes(keyword))) {
  return;
}
```
3 lines instead of 15.